tests/unit/test_clidriver.py

```diff
@@ -279,6 +282,20 @@ def test_ctrl_c_is_handled(self):
         rc = driver.main('s3 list-objects --bucket foo'.split())
         self.assertEqual(rc, 130)
 
+    def test_error_unicode(self):
+        # Nose on Python 2.6 does not support _any_ non ASCII charactes being
+        # written to stdout/stderr, which makes it impossible for this test
+        # to actually succeed on 2.6.
+        if sys.version_info < (2, 7):
+            raise nose.SkipTest
+        driver = CLIDriver(session=self.session)
+        fake_client = mock.Mock()
+        fake_client.list_objects.side_effect = Exception(u"☃")
+        fake_client.can_paginate.return_value = False
+        driver.session.create_client = mock.Mock(return_value=fake_client)
+        rc = driver.main('s3 list-objects --bucket foo'.split())
```

Comment:
So one idea that I have that should solve both skipping the python2.6 test (which we really try to avoid) and upgrading the version of nose is to patch ``sys.stderr`` and then assert the correct exception message got raised to the user? The only reason I suggest this is:

* Even though in the test you are asserting the rc, the exact reason why the rc was set to 255 is unknown unless you take a look at the error printed to standard error.
* While its arguably not the best practice to be patching ``sys.stderr`` for unit tests, it is something it is done all over this test file. So there is a precedent set. For example: https://github.com/aws/aws-cli/blob/develop/tests/unit/test_clidriver.py#L283. What would be better is if the CLIDriver allowed you to inject a file-handle for stdout and stderr so you do not have to patch to assert content written out, but that is unfortunately not the case.

Let me know what you think?